### ISSUE TITLE
Operator whitespace.

### ISSUE THREAD

**Issue by sssbohdan** [realm/SwiftLint#1824]
### Bug Report

Operator Function Whitespace Violation: Operators should be surrounded by a single whitespace when defining them. (operator_whitespace)

```swift
// This triggers a violation:
public static func <(lhs: CGPoint, rhs: CGPoint) -> Bool {
```

**Comment by SDGGiesbrecht** [comment]
This is not a bug; it is precisely what the rule was designed to do. The rule expects the following:

```swift
public static func < (lhs: CGPoint, rhs: CGPoint) -> Bool {}
```

The clarity of this style becomes more apparent when the function signature is more complicated:

```swift
func <<<T : BitShiftable>(lhs: T, rhs: T) -> T {}
//   ?!?
```
vs
```swift
func << <T : BitShiftable>(lhs: T, rhs: T) -> T {}
//     ↑
```

You can always disable the rule if you prefer the unspaced style. In that case, you are not be alone. One reason would be that it is more consistent with non‐operator function declarations, which definitely look odd when they are spaced.

**Comment by sssbohdan** [comment]
@SDGGiesbrecht but this not correct. We can interpret ```<``` as the name of function. Apple not paste space after too.

**Comment by marcelofabri** [comment]
Feel free to disable the rule if you want.

**Comment by marcelofabri** [comment]
i'm closing this because it's not a bug, it's what the rule was meant to do. That rule was based on [the GitHub Swift Style Guide](https://github.com/github/swift-style-guide#use-whitespace-around-operator-definitions), which is the main inspiration for SwiftLint rules.

**Comment by JessyCatterwaul** [comment]
It's a bug because it doesn't respect pre/postfix operators.
